public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -451,7 +455,8 @@ export class LokiDatasource
 
   async getQueryStats(query: string): Promise<QueryStats | undefined> {
     // if query is invalid, clear stats, and don't request
-    if (!isValidQuery(query)) {
+    const errors = validateQuery(query, this.interpolateString(query, placeHolderScopedVars), []);
+    if (typeof errors === 'object') {
       return undefined;
     }
```

Comment:
I regret the way I designed this function, it's really awkward to consume. 

```suggestion
    const errors = validateQuery(query, this.interpolateString(query, placeHolderScopedVars), []);
    if (errors !== false) {
      return undefined;
    }
```

You can also use `Array.isArray()`.

But even better: would you create a new function inside `validation.ts`, with the same signature of `validateQuery`, but it returns a boolean? E.g.

```javascript
function isQueryValid(...): boolean {
  return validateQuery(...) === false ? true : false;
}
```

Reply:
@matyax looking at `validateQuery()`, it doesn't seem like it is used that many places, would it be better to just change the response of this function directly.

Reply:
Good question. I think `validateQuery` is correctly serving its purpose: returning an array of errors. The only change that I would make is to always return an array, empty or with errors.

Then, I think we're missing a function that we (or other developers) can call ot get a boolean indicating if the query is valid or not.